calico_containers/pycalico/util.py

```diff
@@ -10,3 +26,43 @@ def generate_cali_interface_name(prefix, ep_id):
     if len(prefix) > 4:
         raise ValueError('Prefix must be 4 characters or less.')
     return prefix + ep_id[:11]
+
+
+def get_host_ips(version=4, interface=None, exclude=[]):
+    """
+    Gets all IP addresses assigned to this host.
+
+    This function is fail-safe and will return an empty array instead of
+    raising any exceptions.
+
+    :param version: Desired version of IP addresses. Can be 4 or 6. defaults to 4
+    :param interface: string - Desired interface to pull IP address from
+    :param exclude: list of interfaces (strings) to ignore (ex. ["lo","docker0"])
+    :return: List of string representations of IP Addresses.
+    """
+
+    ip_addrs = []
+
+    # Call `ip addr`
+    if interface is None:
+        ip_addr_output = check_output(["ip", "-%d" % (version), "addr"])    
+    else:    # Add argument to `ip addr` command if you are selecting an interface
+        ip_addr_output = check_output(["ip", "-%d" % (version), "addr", "show", "%s" % (interface)])
+
+    # Select Regex for IPv6 or IPv4
+    GET_IP= GET_IPV4 if version==4 else GET_IPV6
+
+    # separate interface blocks from ip addr output and iterate
+    for iface_block in INTERFACE_SPLIT.findall(ip_addr_output):
+        # exclude certain interfaces
+        if GET_IFACE.search(iface_block).group(1) not in exclude :
+            # iterate through Addresses on interface
+            for address in GET_IP.findall(iface_block):
```

Comment:
I don't think dynamically setting GET_IP is very straightforward. It'd be more obvious what's going on if you branched an if statement down here:
```
if version == 4:
    addresses = GET_IPV4.findall(iface_block)
elif verison == 6:
    addresses = GET_IPV6.findall(iface_block)
```